plugins/modules/ldap_auth_provider.py

```diff
@@ -0,0 +1,345 @@
+            type: str
+            default: groupOfNames
+      user_search:
+        description:
+          - Search configuration for users.
+        type: dict
+        suboptions:
+          base_dn:
+            description:
+              - Which part of the directory tree to search.
+            required: true
+            type: str
+          attribute:
+            description:
+              - Used for comparing result entries.
+            type: str
+            default: uid
+          name_attribute:
+            description:
+              - Represents the attribute to use as the entry name.
+            type: str
+            default: cn
+          object_class:
+            description:
+              - Identifies the class of objects returned in the search result.
+            type: str
+            default: person
+  groups_prefix:
+    description:
+      - The prefix added to all LDAP groups.
+    type: str
+  username_prefix:
+    description:
+      - The prefix added to all LDAP usernames.
+    type: str
+
+seealso:
+  - module: sensu.sensu_go.auth_provider_info
+  - module: sensu.sensu_go.ad_auth_provider
+  - module: sensu.sensu_go.oidc_auth_provider
+"""
+
+EXAMPLES = """
+- name: Create a LDAP auth provider
+  sensu.sensu_go.ldap_auth_provider:
+    name: openldap
+    servers:
+      host: 127.0.0.1
+      group_search:
+        base_dn: dc=acme,dc=org
+      user_search:
+        base_dn: dc=acme,dc=org
+
+- name: Delete a LDAP auth provider
+  sensu.sensu_go.ldap_auth_provider:
+    name: openldap
+    state: absent
+"""
+
+RETURN = """
+object:
+  description: Object representing Sensu LDAP authentication provider.
+  returned: success
+  type: dict
+  sample:
+    metadata:
+      name: 'openldap'
+    servers:
+      host: '127.0.0.1'
+      port: '636'
+      insecure: 'False'
+      security: 'tls'
+      trusted_ca_file: '/path/to/trusted-certificate-authorities.pem'
+      client_cert_file: '/path/to/ssl/cert.pem'
+      client_key_file: '/path/to/ssl/key.pem'
+      binding:
+        user_dn: 'cn=binder,dc=acme,dc=org'
+        password: 'YOUR_PASSWORD'
+      group_search:
+        base_dn: 'dc=acme,dc=org'
+        attribute: 'member'
+        name_attribute': 'cn'
+        object_class: 'groupOfNames'
+      user_search:
+        base_dn: 'dc=acme,dc=org'
+        attribute: 'uid'
+        name_attribute: 'cn'
+        object_class: 'person'
+    groups_prefix: 'ldap'
+    username_prefix: 'ldap'
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ..module_utils import arguments, errors, utils
+
+API_GROUP = "enterprise"
+API_VERSION = "authentication/v2"
+
+
+def main():
+    module = AnsibleModule(
+        supports_check_mode=True,
+        argument_spec=dict(
+            arguments.get_spec(
+                "auth",
+                "name",
+                "state",
+            ),
+            servers=dict(
+                type="list",
+                required=True,
```

Comment:
Servers parameter is only required if `state: present`, so we need to handle this with a conditional. See https://github.com/sensu/sensu-go-ansible/blob/master/plugins/modules/asset.py#L161-L165 for example of how we can conditionally check this.